src/ts/index.ts

```diff
@@ -1,32 +1,35 @@
 import checkIfJson from './util/check-if-json';
 import * as ui from './ui';
 import * as $ from 'jquery';
-import {checkStorageOptions} from './options';
+import {defaultOptions, parseOptions, checkStorageOptions} from './options';
 
 let input = '';
 let output = '';
 let filter = '.';
 let port = null;
+let options = defaultOptions;
 
 function onLoad() {
     checkIfJson(function (pre) {
         pre.hidden = true;
 
         checkStorageOptions();
         loadExtension(pre);
+
+        chrome.storage.onChanged.addListener(function (changes) {
```

Comment:
I moved this because the extension actually kept adding a new event listener each time a setting changed. This caused the `loadExtension` function to be called at a doubling rate as you saved options with a tab open.